### ISSUE TITLE
Set Custom Response Headers Feature Request

### ISSUE THREAD

**Issue by gregelenbaas** [begriffs/postgrest#940]
Hi PostgREST team, 

Been reading through the docs and building out a test schema for most of the day. I found how to access request headers but haven't found if it is possible to add response headers from rpc stored procedures. If it's not already a capability, I think it would be really useful. My use case would be for adding Set-Cookie headers that would store the JWT token as well as a second cookie with their user_id and whether they were recognized originally with a password or from an email link. 

Since HTTP request headers seem to be the preferred method for modifying query responses, I think it would make sense for there to be a way to set Response headers as well. Having NGINX or application server code parse the postgREST rpc/authenticate server response before sending to the client seems really inefficient. I also don't want to set the cookie on the client they should be set as HTTP ONLY to prevent client side javascript from sniffing the value of the JWT cookie and trying to hijack the users session. 

Thanks for your consideration of this feature and all your work!

**Comment by ruslantalpa** [comment]
Why would parsing the response in ngxin in order to add the additional header be inefficient?

**Comment by omani** [comment]
@ruslantalpa, although I don't think settings headers in nginx is inefficient, I like the idea of being able to add respond headers within an RPC, because sometimes only within the RPC you know your data and you can set your right headers. nginx (after getting the response back from backend) can set headers yes but more like general purpose headers. not specific, data-releated headers.

**Comment by begriffs** [comment]
This has been a feature we've discussed every now and then. For instance Ruslan once created a pull request to allow stored procs to set cookies, and then we discussed generalizing to setting any kind of response header:

https://github.com/begriffs/postgrest/pull/664#issuecomment-231635405

If one of the approaches in that thread is viable I think it would be a pretty awesome feature. As you mentioned, @omani, the procedure has inside access to any data it needs to determine the headers. Nginx has more limited access to make those decisions.
